acl.go

```diff
@@ -0,0 +1,110 @@
+package imap
+
+import (
+	"fmt"
+	"strings"
+)
+
+// IMAP4 ACL extension (RFC 2086)
+
+type RightSet string
+
+type Right byte
+
+const (
+	RightLookup     = Right('l') // mailbox is visible to LIST/LSUB commands
+	RightRead       = Right('r') // SELECT the mailbox, perform CHECK, FETCH, PARTIAL, SEARCH, COPY from mailbox
+	RightSeen       = Right('s') // keep seen/unseen information across sessions (STORE SEEN flag)
+	RightWrite      = Right('w') // STORE flags other than SEEN and DELETED
+	RightInsert     = Right('i') // perform APPEND, COPY into mailbox
+	RightPost       = Right('p') // send mail to submission address for mailbox, not enforced by IMAP4 itself
+	RightCreate     = Right('c') // CREATE new sub-mailboxes in any implementation-defined hierarchy
+	RightDelete     = Right('d') // STORE DELETED flag, perform EXPUNGE
+	RightAdminister = Right('a') // perform SETACL
+
+	AllRights = RightSet("lrswipcda")
+)
+
+type RightsIdentifier string
+
+const RightsIdentifierAnyone = RightsIdentifier("anyone")
+
+type RightModification byte
+
+const (
+	RightModificationReplace = RightModification(0)
+	RightModificationAdd     = RightModification('+')
+	RightModificationRemove  = RightModification('-')
+)
+
+// NewRights converts rights string into RightModification and RightSet with validation
+func NewRights(rights string, ignoreUnsupported bool) (RightModification, RightSet, error) {
```

Comment:
Do we really need `ignoreUnsupported`?

It's set to `false` in the client code, but I think clients should ignore rights they don't support.

Reply:
I want to be sure I got you right, should we return only standard rights here and ignore nonstandard?

Reply:
I think we should just pass through everything without filtering. Do you see an issue with this, or do you think it would be better to do it differently?